lesson25/src/js/validation.js

```diff
@@ -0,0 +1,67 @@
+const showErrorMessage = (target,message) => {
+    target.classList.add("border-2","border-red-500");
+
+    const errorMessage = document.createElement("p");
+    errorMessage.classList.add("field-invalid","text-sm","my-2","text-rose-600");
+	errorMessage.textContent = message;
+
+	const error = target.nextElementSibling;
+	if(!error){
+		target.insertAdjacentElement("afterend",errorMessage);
+	}else{
+		if(!(error.classList.contains("field-invalid"))){
+			target.insertAdjacentElement("afterend",errorMessage);
+		}
+	}
+}
+
+const removeErrorMessage = (target) => {
+    target.classList.remove("border-red-500");
+
+    const error = target.nextElementSibling;
+    if(error && error.classList.contains("field-invalid")){
+        error.remove();
+    }
```

Comment:
Q.
Since the only thing that deserves "nextElementSibling" in target is the error message, I felt this would still work. Because I think that the error message always have a class of "field-invalid".

```suggestion
    const error = target.nextElementSibling;
    if(error){
        error.remove();
    }
```

Reply:
Thank you for your comment!
If I don't check if it has a field-invalid class, the label will disappear when the checkbox validation checks, so I'm going to leave this one as is.